### DISCUSSION TITLE
Add Support for Non Partitioned Containers

### DISCUSSION THREAD

**Discussion by janrhijn** [IEvangelist/azure-cosmos-dotnet-repository#498] [Ideas]
**Is your feature request related to a problem? Please describe.**
We have some (older) containers which aren't partitioned. We are unable to use this (super handy) package because this is not supported in IEvangelist. In the Microsoft.Azure.Cosmos package itself we can us `PartitionKey.None` when the container isn't partitioned.

**Describe the solution you'd like**
Implement `PartitionKey.None` so it can be used with a container which partitioned, similar as it is used with Microsoft.Azure.Cosmos

**Describe alternatives you've considered**
Overriding `GetPartitionKeyValue()` with `PartitionKey.None.ToString();`. This doesn't work, because this will create a new Item with a partitionkey with value null (which is not the same as no partitionkey at all).

**Comment by marcselman** [comment]
I would love to see this implemented. This should have been supported from the start in my opinion.

**Comment by IEvangelist** [comment]
What would be involved in implementing this? I've been really buried lately, and admittedly have had zero time for side projects, namely this project, but I know lots of folks rely on it. I could make time to review a PR if either of you @janrhijn and/or @marcselman are interested in giving that a try?
